bin/linz_bde_uploader.pl

```diff
@@ -215,7 +219,7 @@
         }
     }
 
-    if($verbose)
+    if($verbose and $log_config and not $dry_run)
```

Comment:
Looking at this I'm now a little confused, we have two place where a stdout loggers is created. Should this not be further up in the conditional logic where it's not a dry run and we are creating an easy STDOUT method?

Reply:
The problem is that we want `-verbose` to possibly _append_ a `stdout` destination to a pre-existing configuration (think configuration file, requesting for example to send logs to email). When using `-dry-run` we are already creating a `stdout` logger, thus the exclusion. Also when NO log configuration is included in the configuration file we are creating a `stdout` logger.

I agree the logic is confusing but this is the best I could come up with at the moment.

A better approach would iterate over the existing logger and ONLY add a `stdout` logger if one is not already found. But the flexibility/complexity of Log4Perl makes that non-easy ( I didn't find an `iterate-all-loggers` function, let alone a `check-if-this-logger-prints-to-stdout`...)

Reply:
Ok yes I understand now.